### PR TITLE
[Harness] Remove the code that check if the test dlls were downloaded.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Initially the location of the test dlls of the downloaded artefact and the compiled mono were different. That was no longer the case and therefore there was a lot of code that could be removed since the wasDownloaded variable was set to be always true.

We remove the not needed code that will calculate a diff path if mono was compiled and use the AssemblyLocator class to point to the correct root path were test dlls will be found.

### CODE REVIEW

File: tests/xharness/BCLTestImporter/BCLTestAssemblyDefinition.cs [hunk 1]
```diff
@@ -53,36 +56,20 @@ public string GetName (Platform platform)
...
 		{
-			if (string.IsNullOrEmpty (downloadsPath))
-				throw new ArgumentNullException (nameof (downloadsPath));
+			var downloadsPath = AssemblyLocator.GetAssembliesRootLocation (platform);
 
 			switch (platform) {
 			case Platform.MacOSFull:
 			case Platform.MacOSModern:
-				return Path.Combine (downloadsPath, "mac-bcl", platformPathMatch [platform], "tests");
+				return Path.Combine (downloadsPath, "mac-bcl",  platformPathMatch [platform], "tests");
```
Comment (mandel-macaque) [inline tests/xharness/BCLTestImporter/BCLTestAssemblyDefinition.cs]:
```suggestion
				return Path.Combine (downloadsPath, "mac-bcl", platformPathMatch [platform], "tests");
```

File: tests/xharness/Xharness.Tests/BCLTestImporter/Tests/TestAssemblyDefinitionTest.cs [hunk 2]
```diff
@@ -1,67 +1,86 @@
...
 		[Test]
 		public void GetPathiOS ()
 		{
-			var testAssemblyDefinition = new BCLTestAssemblyDefinition ("monotouch_System.Json.Microsoft_xunit-test.dll");
+			assemblyLocator.Setup (a => a.GetAssembliesRootLocation (It.IsAny<Platform> ())).Returns (Environment.GetEnvironmentVariable ("HOME"));
+			var testAssemblyDefinition = new BCLTestAssemblyDefinition ("monotouch_System.Json.Microsoft_xunit-test.dll", assemblyLocator.Object);
 			var home = Environment.GetEnvironmentVariable ("HOME");
-			var expectedPath = Path.Combine (home, "mcs/class/lib", "monotouch", "tests", testAssemblyDefinition.Name);
-			Assert.AreEqual (expectedPath, testAssemblyDefinition.GetPath (Environment.GetEnvironmentVariable("HOME"), Platform.iOS, false));
+			var expectedPath = Path.Combine (home, "ios-bcl","monotouch", "tests", testAssemblyDefinition.Name);
```
Comment (mandel-macaque) [inline tests/xharness/Xharness.Tests/BCLTestImporter/Tests/TestAssemblyDefinitionTest.cs]:
```suggestion
			var expectedPath = Path.Combine (home, "ios-bcl", "monotouch", "tests", testAssemblyDefinition.Name);
```

File: tests/xharness/BCLTestImporter/BCLTestAssemblyDefinition.cs [hunk 3]
```diff
@@ -92,28 +79,26 @@ public static string GetTestDirectoryFromDownloadsPath (string downloadsPath, Pl
 			return null;
 		}
 
-		public static string GetHintPathForRefenreceAssembly (string assembly, string monoRootPath, Platform platform, bool isDownload)
+		public static string GetHintPathForRefenreceAssembly (string assembly, string monoRootPath, Platform platform)
```
Comment (premun) [inline tests/xharness/BCLTestImporter/BCLTestAssemblyDefinition.cs]:
```suggestion
		public static string GetHintPathForReferenceAssembly (string assembly, string monoRootPath, Platform platform)
```

File: tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs [hunk 4]
```diff
@@ -11,8 +12,10 @@ namespace Xharness.BCLTestImporter {
 	/// </summary>
 	public partial struct BCLTestProjectDefinition {
 		public string Name { get; set; }
-		public List<BCLTestAssemblyDefinition> TestAssemblies {get; private set;}
 		public string ExtraArgs { get; private set; }
+		public IAssemblyLocator AssemblyLocator { get; set; }
+		public List<BCLTestAssemblyDefinition> TestAssemblies {get; private set; }
```
Comment (premun) [inline tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]:
```suggestion
		public List<BCLTestAssemblyDefinition> TestAssemblies { get; private set; }
```

File: tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs [hunk 5]
```diff
@@ -122,20 +131,20 @@ public bool Validate ()
...
 				throw new ArgumentNullException (nameof (monoRootPath));
 
-			var references = GetProjectAssemblyReferences (monoRootPath, platform, wasDownloaded);
+			var references = GetProjectAssemblyReferences (monoRootPath, platform);
 			if (!string.IsNullOrEmpty (references.FailureMessage))
 				return (references.FailureMessage, null);
 			var asm = references.References.Select (
 					a => (assembly: a, 
-						hintPath: BCLTestAssemblyDefinition.GetHintPathForRefenreceAssembly (a, monoRootPath, platform, wasDownloaded))).Union (
+						hintPath: BCLTestAssemblyDefinition.GetHintPathForRefenreceAssembly (a, monoRootPath, platform))).Union (
```
Comment (premun) [inline tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]:
```suggestion
						hintPath: BCLTestAssemblyDefinition.GetHintPathForReferenceAssembly (a, monoRootPath, platform))).Union (
```

File: tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs [hunk 6]
```diff
@@ -21,22 +24,28 @@ public partial struct BCLTestProjectDefinition {
...
+				TestAssemblies.Add (new BCLTestAssemblyDefinition (assembly, AssemblyLocator));
 			}
 		}
 		
-		public BCLTestProjectDefinition (string name, List<BCLTestAssemblyDefinition> assemblies, string extraArgs)
+		public BCLTestProjectDefinition (string name, IAssemblyLocator locator, List<BCLTestAssemblyDefinition> assemblies, string extraArgs)
 		{
-			Name = name;
-			TestAssemblies = assemblies;
+			Name = name ?? throw new ArgumentNullException (nameof (locator));
```
Comment (spouliot) [inline tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]: wrong `nameof`

File: tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs [hunk 7]
```diff
@@ -21,22 +24,28 @@ public partial struct BCLTestProjectDefinition {
...
+			AssemblyLocator = locator ?? throw new ArgumentNullException (nameof (locator));
+			TestAssemblies = assemblies ?? throw new ArgumentNullException (nameof (locator));
```
Comment (spouliot) [inline tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]: wrong `nameof`

Comment (mandel-macaque) [reply tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]:
```suggestion
			TestAssemblies = assemblies ?? throw new ArgumentNullException (nameof (assemblies));
```

File: tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs (hunk 6 above)
Comment (mandel-macaque) [reply tests/xharness/BCLTestImporter/BCLTestProjectDefinition.cs]:
```suggestion
			Name = name ?? throw new ArgumentNullException (nameof (name));
```

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/1488

**Pull request merged**